File: Atcoder_#333/javascript/b_pentagon.js
function main(input) {
  const lines = input.trim().split('\n');
  const s_input = lines[0].trim();
  const t_input = lines[1].trim();

  const s1 = s_input[0];
  const s2 = s_input[1];
  const t1 = t_input[0];
  const t2 = t_input[1];

  const array = ["A","B","C","D","E"];

  const s1_index = array.indexOf(s1);
  const s2_index = array.indexOf(s2);
  const t1_index = array.indexOf(t1);
  const t2_index = array.indexOf(t2);

  const s_distance = Math.min(
    Math.abs(s1_index - s2_index),
    5 - Math.abs(s1_index - s2_index)
  );
  const t_distance = Math.min(
    Math.abs(t1_index - t2_index),
    5 - Math.abs(t1_index - t2_index)
  );

  if (s_distance == t_distance) {
    console.log("Yes");
  } else {
    console.log("No");
  }
}

main(require("fs").readFileSync("/dev/stdin", "utf8"));

File: Atcoder_#333/ruby/b_pentagon.rb
s_input = gets.chomp
t_input = gets.chomp

s1, s2 = s_input[0], s_input[1]
t1, t2 = t_input[0], t_input[1]

array = ["A", "B", "C", "D", "E"]

s1_index = array.index(s1)
s2_index = array.index(s2)
t1_index = array.index(t1)
t2_index = array.index(t2)

s_distance = [ (s1_index - s2_index).abs, (5 - (s1_index - s2_index).abs) ].min
t_distance = [ (t1_index - t2_index).abs, (5 - (t1_index - t2_index).abs) ].min

if s_distance == t_distance
  puts "Yes"
else
  puts "No"
end
